### PR TITLE
Business intelligence

### PR DESCRIPTION
**Pull request by ahenzler**
Created the business intelligence methods

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -107,6 +108,8 @@ def merchants_with_high_item_count
     end
     merchant_ids_with_high_item_count.map do |merchant_id|
       @merchants_repo.find_by_id(merchant_id)
+      #where you are iterating more than once in one method
```
Comment (jrwhitmer) [inline lib/sales_analyst.rb:111]: Thank you for these refactor comments, I will look into reducing nested iteration in this class

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -169,6 +173,7 @@ def top_merchants_by_invoice_count
       @merchants_repo.find_by_id(merchant_id)
     end
   end
+  #stub helpers allow them to recive helper and tell it
```
Comment (jrwhitmer) [inline lib/sales_analyst.rb:176]: Do you mean stubbing within the class or in the test?

**Review by jrwhitmer — APPROVED** [review]
These methods look good!

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -224,4 +229,47 @@ def invoice_status(status)
...
+  def find_transaction(id)
+    all_transactions.find_all do |transaction|
+      transaction.invoice_id == id
+    end
+  end
+
+  def transactions_exist?(invoice_id)
+    @transactions = find_transaction(invoice_id)
+    @transactions != []
+  end
```
Comment (deebot10) [inline lib/sales_analyst.rb:242]: This was a great way. to find invoice ids that don't exists in the list

**Review by deebot10 — APPROVED** [review]

**Review by rcasias — APPROVED** [review]
Looks great! Great work Alicia!

**Pull request merged**